test/file_format_t.cpp

```diff
@@ -1306,6 +1311,234 @@ void FileFormatTest::ocdTextImportTest()
+		    { C(100), C(-250) },
+		    { C(150), C(-260) },
+		    { C(100), C(-250) },  // same as first point
+		    { C(120), C(-200) | Ocd::OcdPoint32::FlagHole },
+		    { C(200), C(-350) | Ocd::OcdPoint32::FlagHole },
+		    { C(220), C(-400) },
+		    { C(200), C(-350) }  // same as second FlagHole point
+		};
+		static int expected_flags_area[] = {
+		    0,
+		    0,
+		    MapCoord::ClosePoint | MapCoord::HolePoint,
+		    MapCoord::ClosePoint | MapCoord::HolePoint,
+		    0,
+		    0,
+		    MapCoord::ClosePoint
+		};
+		QTest::newRow("empty hole, area") << OcdPointsView(ocd_points) << Area << FlagsView(expected_flags_area);
+		static int expected_flags_line[7] = {};
+		QTest::newRow("empty hole, line") << OcdPointsView(ocd_points) << Line << FlagsView(expected_flags_line);
+	}
+	
+	{
+		// straight segments, with two "empty" holes
+		static Ocd::OcdPoint32 ocd_points[] = {
+		    { C(100), C(-250) },
+		    { C(150), C(-260) },
+		    { C(100), C(-250) },  // same as first point
+		    { C(120), C(-200) | Ocd::OcdPoint32::FlagHole },
+		    { C(140), C(-300) | Ocd::OcdPoint32::FlagHole },
+		    { C(200), C(-350) | Ocd::OcdPoint32::FlagHole },
+		    { C(220), C(-400) },
+		    { C(200), C(-350) },  // same as third FlagHole point
+		};
+		static int expected_flags_area[] = {
+		    0,
+		    0,
+		    MapCoord::ClosePoint | MapCoord::HolePoint,
+		    MapCoord::ClosePoint | MapCoord::HolePoint,
+		    MapCoord::ClosePoint | MapCoord::HolePoint,
+		    0,
+		    0,
+		    MapCoord::ClosePoint
+		};
+		QTest::newRow("two empty holes, area") << OcdPointsView(ocd_points) << Area << FlagsView(expected_flags_area);
+		static int expected_flags_line[8] = {};
+		QTest::newRow("two empty holes, line") << OcdPointsView(ocd_points) << Line << FlagsView(expected_flags_line);
+	}
+	
+	{
+		// straight segments, one hole, not actual ares
+		static Ocd::OcdPoint32 ocd_points[] = {
+		    { C(100), C(-250) },
+		    { C(150), C(-260) },
+		    { C(200), C(-350) | Ocd::OcdPoint32::FlagHole },
+		    { C(220), C(-400) }
+		};
+		static int expected_flags_area[] = {
+		    0,
+		    0,
+		    MapCoord::ClosePoint | MapCoord::HolePoint,  // injected by Mapper
+		    0,
+		    0,
+		    MapCoord::ClosePoint  // injected by Mapper
+		};
+		QTest::newRow("open areas with hole, area") << OcdPointsView(ocd_points) << Area << FlagsView(expected_flags_area);
+		static int expected_flags_line[4] = {};
+		QTest::newRow("open areas with hole, line") << OcdPointsView(ocd_points) << Line << FlagsView(expected_flags_line);
+	}
+}
+
+void FileFormatTest::ocdPathImportTest()
+{
+	QFETCH(OcdPointsView, points);
+	QFETCH(OcdPathPersonality, personality);
+	QFETCH(FlagsView, expected);
+	
+	TestOcdFileImport ocd_v12_import{12};
+	
+	TestOcdFileImport::OcdImportedPathObject path_object;
+	ocd_v12_import.fillPathCoords(&path_object, personality == Area, points.size, points.data);
+	QVERIFY(path_object.getRawCoordinateVector().size() > 0);
```

Comment:
There was a time when `OcdPointsView` and `FlagsView` returned empty views, turning the test into a no-op.